File: chapter_4/vb/code_1/code3_protected.vb
Module code3_protected
    Public Class Employee
        Public empName As String
        Public empID As Integer = 100
        Protected carPay As Single
        Protected empAge As Integer
        Protected Shared companyName As String

        Sub m()
            carPay = 678
            Console.WriteLine(carPay)
        End Sub
    End Class

    Class manager
        Inherits Employee
        Sub n()
            Console.WriteLine(empAge & vbCrLf)
        End Sub


    End Class


    Sub Main()
        Dim objEmp As New Employee
        Dim objMan As New manager
        objEmp.m()
        objMan.n()
    End Sub
End Module

File: chapter_4/Csharp/Csharp/code3_protected.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Csharp
{
    class Employee
    {
        public string empName ;
        public int empID  = 100;
        protected Single carPay;
        protected int empAge ;
        protected string companyName ;

        public void m()
        {
            carPay = 1008;
            Console.WriteLine(carPay);
        }
    }

    class manager:Employee
    {
        public void n()
        {
            Console.WriteLine(empAge + "\n");
        }

    }
    class code3_protected
    {
        public static void display()
        {
            Employee objEmp = new Employee();
            manager objMan = new manager();
            objEmp.m();
            objMan.n();
        }
    }
}
